pinot-core/src/test/java/org/apache/pinot/queries/TextSearchQueriesTest.java

```diff
@@ -280,6 +282,34 @@ private List<GenericRow> createTestData()
     return rows;
   }
 
+  @Test
+  public void testMultiTermRegexSearch()
+      throws Exception {
+    // Search in SKILLS_TEXT_COL column to look for documents that have the /.*ealtime stream system.*/ regex pattern
+    List<Object[]> expected = new ArrayList<>();
+    expected.add(new Object[]{1010,
+        "Distributed systems, Java, realtime streaming systems, Machine learning, spark, Kubernetes, distributed "
+            + "storage, concurrency, multi-threading"});
+    expected.add(new Object[]{1019,
+        "C++, Java, Python, realtime streaming systems, Machine learning, spark, Kubernetes, transaction processing, "
+            + "distributed storage, concurrency, multi-threading, apache airflow"});
+
+    String query =
+        "SELECT INT_COL, SKILLS_TEXT_COL FROM MyTable WHERE TEXT_MATCH(SKILLS_TEXT_COL, '*ealtime streaming system*') "
+            + "LIMIT 50000";
```

Comment:
Can we handle/add a test case for boolean span queries? i.e. 
```
TEXT_MATCH(SKILLS_TEXT_COL, '*ealtime streaming system* AND *chine learn*') 
```

Reply:
Test cases added and they work! Though it is not the main intent of this PR.